### PR TITLE
wifi: Fix clock include

### PR DESCRIPTION
**Pull request by krish2718**
Build fixes.

### CODE REVIEW

**Review by muraliThokala — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

File: samples/wifi/scan/src/main.c [hunk 1]
```diff
@@ -217,7 +219,7 @@ int main(void)
 
 	net_mgmt_add_event_callback(&wifi_shell_mgmt_cb);
 
-#ifdef CLOCK_FEATURE_HFCLK_DIVIDE_PRESENT
+#if defined(CLOCK_FEATURE_HFCLK_DIVIDE_PRESENT) || NRF_CLOCK_HAS_HFCLK192M
```
Comment (ajayparida) [inline samples/wifi/scan/src/main.c:222]: Better to use
if defined(CLOCK_FEATURE_HFCLK_DIVIDE_PRESENT) || defined(NRF_CLOCK_HAS_HFCLK192M)
Apllicable for rest of the files in this PR.

Comment (krish2718) [reply samples/wifi/scan/src/main.c:222]: Oops, its merged, but this is the way its used elsewhere (See: `modules/hal/nordic_haltium/nrfx/drivers/include/nrfx_clock.h`), don't really know why `defined` is skipped though.

Comment (bama-nordic) [reply samples/wifi/scan/src/main.c:222]: I really think `defined` should not be skipped - @krish2718 maybe you can raise a PR with this correction?

Comment (krish2718) [reply samples/wifi/scan/src/main.c:222]: Wanted to keep the same conditions across NCS.

Comment (krish2718) [reply samples/wifi/scan/src/main.c:222]: Here's the reason, it shouldn't be just `defined()` as the condition should only be included if its `1`, see https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/hal/nrf_clock.h#L76

**Pull request merged**